### ISSUE TITLE
[Feature]: Only authenticated users should be able to list domains

### ISSUE THREAD

**Issue by gcapizzi** [cloudfoundry/cf-k8s-controllers#584]
### Blockers

* #721 

### Background

**As an** admin
**I want** to assign the `SpaceDeveloper` role to the users I trust
**So that** they can push apps to their spaces

### Acceptance Criteria

**GIVEN** a [shared domain](https://docs.cloudfoundry.org/devguide/deploy-apps/routes-domains.html#shared-domains) `shared.domain` exists
- **WHEN I** [list domains](https://v3-apidocs.cloudfoundry.org/version/3.113.0/index.html#list-domains) as an authenticated user
  **THEN I** see `shared.domain` in the list
- **WHEN I** list domains without providing an authentication header
  **THEN I** get the following response:
  ```
  HTTP/1.1 401 Unauthorized
  Content-Type: application/json
  ```
  ```json
  {
    "errors": [
      {
        "code": 10002,
        "title": "CF-NotAuthenticated",
        "detail": "Authentication error"
      }
    ]
  }
  ```
- **WHEN I** list domains providing an invalid authentication header
  **THEN I** get the following response:
  ```
  HTTP/1.1 401 Unauthorized
  Content-Type: application/json
  ```
  ```json
  {
    "errors": [
      {
        "code": 1000,
        "title": "CF-InvalidAuthToken",
        "detail": "Invalid Auth Token"
      }
    ]
  }
  ```

### Dev notes

This story will already pass acceptance due to our [authentication middleware](https://github.com/cloudfoundry/cf-k8s-controllers/blob/main/api/apis/authentication_middleware.go). We still want to implement user authentication at the endpoint level, as [this check](https://github.com/cloudfoundry/cf-k8s-controllers/blob/152114ff87b154b072666c0f59c84a9dc127be91/api/apis/authentication_middleware.go#L64-L74) will eventually be removed. I will accept the story by commenting out the middleware.

**Comment by kieron-dev** [comment]
We might not be in a position to do this yet:
- There is only a list domains endpoint. We cannot create a domain with the API yet
- The deploy-on-kind script creates a single cfdomain called 'vcap.me'
- cfdomain is not namespaced

For authz to work on domains, they must be namespaced entities as we cannot give cluster-wide permissions to a resource. So adding a blocked label for now and I'll see if anyone on the channel has more information.

**Comment by gcapizzi** [comment]
Given [private domains are not a priority for now](https://github.com/cloudfoundry/cf-k8s-controllers/issues/663#issuecomment-1055861208), I have rephrased this story to only mention shared domains.

**Comment by gcapizzi** [comment]
@kieron-dev re: testing this without a creation endpoint, can we do what #663 is doing (for now) and unblock this?

**Comment by kieron-dev** [comment]
We have a createDomain() helper in the e2e suite, so fine.

Just wondering whether this story could be merged into #721. And I'm not sure how #663 can work without #721, as shared domains are in the root namepsace, and space-developer will not give you any access there.

**Comment by gcapizzi** [comment]
I thought about it but then I still preferred the two pieces of work to be separate and just block. We might have to block #663 too though: @Birdrock @tcdowney @gnovv wdyt?
